tests/Keboola/ElasticsearchWriter/RunTest.php

```diff
@@ -102,7 +102,50 @@ public function testRunAction()
 
 		$this->assertEquals(0, $returnCode);
 		$this->assertRegExp('/Elasticsearch writer finished successfully/ui', $lastOutput);
-		$this->assertCount(6, $output);
+
+		$this->assertEquals(1, $this->parseBatchCountFromOutput($output));
+	}
+
+	public function testRunWithBulkSizeAction()
+	{
+		$config = [
+			"parameters" => [
+				"elastic" => [
+					"host" => getenv('EX_ES_HOST'),
+					"port" => getenv('EX_ES_HOST_PORT'),
+					"bulkSize" => 30
+				],
+				"tables" => [
+					[
+						"file" => "language-large.csv",
+						"index" => $this->index,
+						"type" => "language",
+						"id" => "id",
+						"export" => true,
+					]
+				]
+			]
+		];
+
+		$yaml = Yaml::dump($config);
+
+		$inTablesDir = './tests/data/run/in/tables';
+		if (!is_dir($inTablesDir)) {
+			mkdir($inTablesDir, 0777, true);
+		}
+
+		file_put_contents('./tests/data/run/config.yml', $yaml);
+
+		copy('./tests/data/csv/language-large.csv', $inTablesDir . '/language-large.csv');
+
+		$lastOutput = exec('php ./src/run.php --data=./tests/data/run', $output, $returnCode);
+
+		$this->assertEquals(0, $returnCode);
+		$this->assertRegExp('/Elasticsearch writer finished successfully/ui', $lastOutput);
+
+		$expectedBatchCount = ceil($this->countTable(new CsvFile($inTablesDir . '/language-large.csv')) / 30);
+		$this->assertEquals($expectedBatchCount, $this->parseBatchCountFromOutput($output));
```

Comment:
to mohlo asi klidně zůstat u toho jednoduchýho countu, ale 👍 za pečlivost :-)

Reply:
já jen pro jistotu